x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -60,6 +60,114 @@ func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx
 		SwapFee:                   sdk.NewDecWithPrec(1, 2),
 	}, nil
 }
+func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
+	// get random pool
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, tokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we have atleast 2 tokens
+	if len(tokens) < 2 {
+		return nil, fmt.Errorf("not enough pool denoms tokens, require 2 got: %d", len(tokens))
+	}
+
+	// Randomize tick values from minTick to maxTick
+	minTick, maxTick := clkeeper.GetMinAndMaxTicksFromExponentAtPriceOne(clPool.GetPrecisionFactorAtPriceOne())
+
+	// Randomize lowerTick and upperTick to create position
+	lowerTick, upperTick, err := getRandomTickPositions(minTick, maxTick, clPool.GetTickSpacing())
+	if err != nil {
+		return nil, err
+	}
+
+	return &cltypes.MsgCreatePosition{
+		PoolId:        clPool.GetId(),
+		Sender:        sender.Address.String(),
+		LowerTick:     lowerTick,
+		UpperTick:     upperTick,
+		TokenDesired0: tokens[0],
+		TokenDesired1: tokens[1],
+		// TODO: Randomize TokenMinAmount0 and TokenMinAmount1 in next iteration
+		TokenMinAmount0: sdk.NewInt(0),
+		TokenMinAmount1: sdk.NewInt(0),
+	}, nil
+}
+
+func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
+	// get random pool
+	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	positions, err := k.GetUserPositions(ctx, sender.Address)
+	if err != nil {
+		return nil, fmt.Errorf("position does not exist")
+	}
+
+	if len(positions) == 0 {
+		return nil, fmt.Errorf("user does not have any position")
+	}
+
+	// pick a random position
+	randPosition := positions[rand.Intn(len(positions))]
+
+	withdrawAmount := randPosition.Liquidity.TruncateInt()
```

Comment:
This is a good change from last time, but can we randomize the withdraw amount to be less than or equal to the total liquidity of the position?

Reply:
I tried this before, but the withdrawAmt gets so large sometimes that it gets out of Int64 bounds. For ex: `256665256162150952146648277` int64() out of bound. But we accept this in sdk.Int 

I found a new way where i just withdraw a random % from 0-1 of total liquidity